src/components/dashboard/components/dictionary/common/DictionaryModal.jsx

```diff
@@ -131,6 +165,9 @@ export class DictionaryModal extends React.Component {
                       placeholder="English"
                       onChange={this.onChange}
                     >
+                      {isEditingDictionary &&
+                        this.props.preSelectDefaultLocale()
+                      }
```

Comment:
```
{isEditingDictionary &&
   this.props.preSelectDefaultLocale()
}
```
Calling the function `preSelectDefaultLocale` in the render method is not really a good idea, this is becuase the function would be called again every single time this component re-renders.
The function `preSelectDefaultLocale` is looping through a preety large array:
```
preSelectDefaultLocale = () => {
  ...
  for (let i = 0; i < languages.length; i++) { // eslint-disable-line
    ...
  }
  ...
}
```
Calling it multiple times like this is not efficient.
I have two suggestion to improve this piece of functionality:
- Since the result of `preSelectDefaultLocale` is not dependant on any data from this component(`DictionaryModal`), you can just pass the result of `preSelectDefaultLocale` from `EditDictionary.jsx` to this component.
So in `EditDictionary.jsx`, you would have something like this:
```
 <DictionaryModal
   ...
   // before -> preSelectDefaultLocale={this.preSelectDefaultLocale}  
   defaultLocate= {preSelectDefaultLocale()}
   ...
 />
```
- The second suggestion is to keep the function call `preSelectDefaultLocale` outside the render method.

Reply:
Thanks for the feedback. However, it seems your suggestions doesn't really fit the context of what I achieved using that. That method preselects a dropdown with an item in the array for the sake of improving user experience, this item will be needed before rendering completes. Speaking of efficiency, will you consider Array.map 
 in the render method inefficient? Beside the array doesn't seem large for the speed of a computer.

Reply:
Thanks again for the feedback. I will look into this.

Reply:
I've updated the code.

Reply:
Nice one 👍.